vsm

```diff
@@ -314,6 +331,73 @@ def run(state):
 def get_runtime():
     return round(time.perf_counter() * 1000 - start_time_ms)
 
+class LogReplayer(object):
```

Comment:
Cosmetic nitpick, it'd be nice to move up this class and group it with the other classes too.